plugins/module_utils/mh/mixins/dest.py

```diff
@@ -0,0 +1,369 @@
+    }
+
+    def __init__(self, module=None):
+        # type: (dict | AnsibleModule | None) -> None
+        self.raw_content = None  # type: bytes
+        self._dest_check = None  # type: DestFileChecks
+        self._dest_infos = None  # type: dict
+        self._dest_attrs = None   # type: dict
+        super(DestFileMixin, self).__init__(module)
+
+    def __init_module__(self):
+        # type: () -> None
+        """
+        Checks and get some informations about the destination file.
+        """
+        super(DestFileMixin, self).__init_module__()
+        self.vars.set('created', False)
+        self._dest_check = DestFileChecks(self.vars['path'])
+        self._dest_check.check(self.vars['allow_creation'], self.vars['backup'])
+        self._dest_infos = self._dest_check.infos()
+        self._dest_attrs = self.module.load_file_common_arguments(self.module.params)
+
+    def __run__(self):
+        # type: () -> None
+        self.__read_dest__()
+        self.__process_data__()
+
+    def __read_dest__(self):
+        # type: () -> None
+        """
+        Open the destination file and put its binary raw contents in
+        `self.raw_content`.
+        """
+        try:
+            with open(self.vars['path'], 'rb') as file:
+                self.raw_content = file.read()
+        except ERR_FILE_NOT_FOUND_OPEN:
+            pass
+
+    def __process_data__(self):
+        # type: () -> None
+        """
+        Implement this in your module to define how to process data the
+        destination file.
+
+        When you run this method, you get binary raw data from your destination
+        file into `self.raw_content` method.
+
+        After you process it, do not write the result directly in the
+        destination file (it is automatically done later with the
+        content of `self.raw_content`) instead, put your result in binary form
+        `self.raw_content`.
+
+        It is expected that this method manage the module change flag if data
+        was updated. A good way do de this is by adding an item in `self.vars`
+        which can handle diffs and changes on data.
+        """
+        raise NotImplementedError()
+
+    def __quit_module__(self):
+        # type: () -> None
+        self.__update_dest__()
+
+    def __update_dest__(self):
+        # type: () -> None
+        if self.has_changed():
+            self._write_temp_file()
+            self._backup_dest()
+            self._replace_dest_with_temp()
+            self.vars.set('created', self._dest_infos['can_create'])
+        self._update_dest_fs_attributes()
+
+    @check_mode_skip
+    def _write_temp_file(self):
+        # type: () -> None
+        """
+        Write the binary data from `self.raw_content` into an Ansible temp
+        file.
+        """
+        fd, self._tmpfile = tempfile.mkstemp(dir=self.module.tmpdir)
+        os.write(fd, self.raw_content)
+        self.module.add_cleanup_file(self._tmpfile)
+        os.close(fd)
```

Comment:
Safer as:
```suggestion
        try:
            fd, self._tmpfile = tempfile.mkstemp(dir=self.module.tmpdir)
            os.write(fd, self.raw_content)
        finally:
            self.module.add_cleanup_file(self._tmpfile)
            os.close(fd)
```

Reply:
@russoz 

Sorry, I'm not reactive to respond.

I need to explicitly update meta directly on the `VarMeta` object else it won't work. It's because if meta already exists in an `VarDict` object, `set` always reuse the value of current meta. Currently, it's not possible to update meta with `VarMeta::set`.

Current `set` method in `VarDict` :
```python
def set(self, name, value, **kwargs):
    if name in ('_data', '_meta'):
        raise ValueError("Names _data and _meta are reserved for use by ModuleHelper")
    self._data[name] = value
    if name in self._meta:        # <- Re-use current meta if already exists
        meta = self.meta(name)
    else:
        meta = VarMeta(**kwargs)
    meta.set_value(value)
    self._meta[name] = meta
```

I have re-based master and fixed the others suggests.